$NetBSD: patch-ad,v 1.1 2008/01/09 14:10:07 martti Exp $

Check for strcasestr. We provide an implementation for it if needed.

--- common/smtppass.c.orig	2006-08-30 20:45:57.000000000 +0300
+++ common/smtppass.c	2007-12-31 12:45:35.075661000 +0200
@@ -206,6 +206,13 @@
 static const char* get_successful_rsp(const char* line, int* cont);
 static void do_server_noop(spctx_t* ctx);
 
+/*------------------------------------------------------
+ *Added to support systems without strcasestr
+ */
+#ifndef HAVE_STRCASESTR
+char *strcasestr(const char *big, const char *little);
+#endif
+ 
 /* Used externally in some cases */
 int sp_parse_option(const char* name, const char* option);
 
@@ -2069,3 +2076,20 @@
     return 0;
 }        
 
+#ifndef HAVE_STRCASESTR
+char *strcasestr(const char *big, const char *little)
+{
+    int blen;
+    int llen;
+    blen = strlen(big);
+    llen = strlen(little);
+    while (blen >= llen) {
+        if (strncasecmp(big, little, llen) == 0)
+            return (char *)big;
+        big++;
+        blen--;
+    }
+    
+    return NULL;
+}
+#endif
